client/components/SurrenderForm.js

```diff
@@ -0,0 +1,187 @@
+      }
+      const responseBody = await response.json()
+      console.log("posted succesfully", responseBody)
+      setMessage("Your surrender request is in process.")
+      setErrors({})
+      clearForm()
+    } catch (error) {
+      console.error("error in form fetch POST", error)
+    }
+  }
+
+  const isValid = () => {
+    let submitErrors = {}
+    const requiredFields = ['personName', 'phoneNumber', 'email', 'petName', 'age', 'petTypeId', 'imgUrl', 'vaccinationStatus', 'adoptionStory']
+    requiredFields.forEach(field => {
+      if (surrender[field].trim() === "") {
+        submitErrors = {
+          ...submitErrors,
+          [field]: "is blank"
+        }
+      }
+    })
+
+    setErrors(submitErrors)
+    return Object.entries(submitErrors).length === 0 && submitErrors.constructor === Object
+  }
+
+  const clearForm = () => {
+    setSurrender({
+      personName: "",
+      phoneNumber: "",
+      email: "",
+      petName: "",
+      age: "",
+      petTypeId: "",
+      imgUrl: "",
+      vaccinationStatus: "",
+      adoptionStory: "",
+    })
+    setErrors({})
+  }
+
+  return (
+    <div>
+      <ErrorList errors={errors} />
+      <h4>{message}</h4>
+      <form onSubmit={submitHandler}>
+        <div>
+          <label htmlFor="personName">Name:
+            <input
+              type="text"
+              id="personName"
+              name="personName"
+              onChange={handleInputChange}
+              value={surrender.personName}
+            />
+          </label>
+
+          <label htmlFor="phoneNumber">Phone Number:
+            <input
+              type="text"
+              id="phoneNumber"
+              name="phoneNumber"
+              onChange={handleInputChange}
+              value={surrender.phoneNumber}
+            />
+          </label>
+
+          <label htmlFor="email">Email:
+            <input
+              type="text"
+              id="email"
+              name="email"
+              onChange={handleInputChange}
+              value={surrender.email}
+            />
+          </label>
+
+          <label htmlFor="petName">Pet Name:
+            <input
+              type="text"
+              id="petName"
+              name="petName"
+              onChange={handleInputChange}
+              value={surrender.petName}
+            />
+          </label>
+
+          <label htmlFor="age">Age:
+            <input
+              type="text"
+              id="age"
+              name="age"
+              onChange={handleInputChange}
+              value={surrender.age}
+            />
+          </label>
+
+          <label htmlFor="petTypeId">Pet Type:
+            <select name="petTypeId" id="petTypeId" onChange={handleInputChange}>
+              <option value=""></option>
+              <option value="2">cat</option>
+              <option value="1">dog </option>
+              <option value="3">mythical creature</option>
+            </select>

```

Comment:
Instead of hardcoding this lets do it dynamically! There is already an endpoint that we can set up a useEffect to fetch there which will return all the pet types.  Then we can store that in state and map through it to generate each `<option>` element

Reply:
Great feedback and tips of how to approach this!  We have refactored accordingly